### PR TITLE
feat: add utility for menu styles

### PR DESCRIPTION
**Pull request by jzempel**
## Description

With the new `menuStyles` utility, menus in Garden are now clearly:
1. A positioning wrapper (facilitated via Popper)
2. A styled menu child

### Features

- New `menuStyles` utility added to `react-theming` and shared between `react-datepickers` and `react-dropdowns`
- Ability to target menu wrappers with themed component custom styling

### Fixes

- Added visibility controls (via `useEffect` with timeouts) to restore fade-out animation for both datepicker and dropdown menus –  providing subtle, important feedback that a user's action was effective
- Implemented CSS that cut out unnecessary "div soup" in dropdown menus.
  - Hiding arrow border during first phase of animation got rid of `StyledMenuAnimation` div.
  - Overriding menu positioning to `static` allows arrow to display beyond `overflow-y` and got rid of `StyledMaxHeightWrapper` div.
- Restored semantic `<ul>` - `<li>` menu structure in dropdowns.

### Other

- Document no-JSX expectation for view components. This cleans up code clutter and, in this PR, improved bundle sizing.

## Detail

Demo site is pre-published for review. Notable links include:
- [theming](https://musing-wright-9e066e.netlify.com/theming/#menustyles) with knob controls to demonstrate the new `menuStyles` functionality
- [datepickers](https://musing-wright-9e066e.netlify.com/datepickers/#default-usage) updated to ensure the calendar dismiss fades out as expected
- [dropdowns](https://musing-wright-9e066e.netlify.com/dropdowns/#menu-usage) updated to ensure the menu dismiss fades out as expected

Merge of this PR will be paired with #651 to publish `react-components` v8.1.

## Checklist

- [x] :ok_hand: design updates are Garden Designer approved (add the
      designer as a reviewer)
- [x] :globe_with_meridians: Styleguidist demo is up-to-date (`yarn start`)
- [x] :arrow_left: renders as expected with reversed (RTL) direction
- [x] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [x] :guardsman: includes new unit tests
- [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

File: packages/theming/src/utils/arrowStyles.ts [hunk 1]
```diff
@@ -30,9 +30,13 @@ export type ARROW_OPTIONS = {
 
 const animationStyles = (position: ARROW_POSITION, modifier: string) => {
   const property = position.split('-')[0];
+  /**
+   * 1. Prevent bleed-through on revised stacking context (i.e. parent transform)
+   */
   const animationName = keyframes`
     0%, 66% {
       ${property}: 2px;
+      border: transparent; /* [1] */
```
Comment (austin94) [inline packages/theming/src/utils/arrowStyles.ts]: So this was the trick when I first moved over to `transform: translate` for the animation. Good find.

File: packages/dropdowns/src/elements/Menu/Menu.tsx [hunk 2]
```diff
@@ -106,12 +123,37 @@ const Menu: React.FunctionComponent<IMenuProps & ThemeProps<DefaultTheme>> = pro
...
+              <StyledMenu
+                isCompact={isCompact}
+                maxHeight={maxHeight}
+                style={computedStyle}
+                {...menuProps}
+              >
+                {(isOpen || isVisible) && children}
+              </StyledMenu>
+            </StyledMenuWrapper>
+            /*
```
Comment (austin94) [inline packages/dropdowns/src/elements/Menu/Menu.tsx]: Can this comment be removed?

File: packages/datepickers/src/styled/StyledMenu.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        <StyledMenuView placement={placement} isAnimated={isAnimated}>
+          {children}
+        </StyledMenuView>
+      </StyledMenuWrapper>
+    );
+  }
+);
+
+StyledMenu.displayName = 'StyledMenu';
+*/
```
Comment (austin94) [inline packages/datepickers/src/styled/StyledMenu.ts]: Extra comment

File: packages/datepickers/src/elements/Datepicker/Datepicker.tsx [hunk 4]
```diff
@@ -200,11 +223,33 @@ const Datepicker: React.FunctionComponent<IDatepickerProps & ThemeProps<DefaultT
...
+                  <Calendar
+                    isCompact={isCompact}
+                    value={value}
+                    minValue={minValue}
+                    maxValue={maxValue}
+                    locale={locale}
+                  />
+                </StyledMenu>
+              </StyledMenuWrapper>
+              /*
```
Comment (austin94) [inline packages/datepickers/src/elements/Datepicker/Datepicker.tsx]: Extra comment

**Comment by jzempel** [comment]
Nice finds @ginnywood. I'm going to add an issue for us to treat these as follow-on bugs 🐛

**Pull request merged**